src/Nfts/Nfts.tsx

```diff
@@ -1,70 +1,84 @@
 import {useNavigation} from '@react-navigation/native'
-import React from 'react'
+import React, {ReactNode} from 'react'
 import {defineMessages, useIntl} from 'react-intl'
 import {RefreshControl, ScrollView, StyleSheet, Text, View} from 'react-native'
 import {SafeAreaView} from 'react-native-safe-area-context'
 
 import {Icon, Spacer} from '../components'
-import {useNfts} from '../hooks'
 import {WalletStackRouteNavigation} from '../navigation'
-import {useSelectedWallet} from '../SelectedWallet'
+import {useFilteredNfts} from './hooks'
 import {ImageGallery, SkeletonGallery} from './ImageGallery'
 import NoNftsScreen from './NoNftsScreen'
 
-type Props = {
-  search?: string
-}
-
-export const Nfts = ({search}: Props) => {
-  const searchTermLowerCase = (search ?? '').toLowerCase()
-  const wallet = useSelectedWallet()
-  const {nfts, isLoading, refetch, isRefetching, isError} = useNfts(wallet)
+export const Nfts = () => {
   const navigation = useNavigation<WalletStackRouteNavigation>()
-  const filteredNfts =
-    searchTermLowerCase.length > 0 && nfts.length > 0
-      ? nfts.filter((n) => n.name.toLowerCase().includes(searchTermLowerCase))
-      : nfts
+  const {search, filteredNfts, isLoading, refetch, isRefetching, isError} = useFilteredNfts()
 
   const navigateToDetails = (id: string) =>
     navigation.navigate('nft-details-routes', {screen: 'nft-details', params: {id}})
-  const handleNftSelect = (index: number) => navigateToDetails(nfts[index].id)
+  const handleNftSelect = (index: number) => navigateToDetails(filteredNfts[index].id)
+
+  if (isError) {
+    return (
+      <ScreenWrapper>
+        <ErrorScreen onRefresh={refetch} isRefreshing={isRefetching} />
+      </ScreenWrapper>
+    )
+  }
+
+  if (isLoading) {
+    return (
+      <ScreenWrapper>
+        <LoadingScreen nftsCount={filteredNfts.length} onRefresh={refetch} isRefreshing={isRefetching} />
+      </ScreenWrapper>
+    )
+  }
+
+  if (search.length > 0 && filteredNfts.length === 0) {
+    return (
+      <ScreenWrapper>
+        <ScrollView
+          style={styles.scrollView}
+          contentContainerStyle={styles.scrollViewError}
+          refreshControl={<RefreshControl onRefresh={refetch} refreshing={isRefetching} />}
+        >
+          <NoNftsScreen />
+        </ScrollView>
+      </ScreenWrapper>
+    )
+  }
+
+  if (search.length === 0 && filteredNfts.length === 0) {
+    return (
+      <ScreenWrapper>
+        <ScrollView
+          style={styles.scrollView}
+          contentContainerStyle={styles.scrollViewError}
+          refreshControl={<RefreshControl onRefresh={refetch} refreshing={isRefetching} />}
+        >
+          <NoNftsScreen count={<NftCount count={filteredNfts.length} />} />
+        </ScrollView>
+      </ScreenWrapper>
+    )
+  }
```

Comment:
these two states for `NoNftsScreen` are looking so similar, maybe the design could update the screen to contain a message when there are no NFTS for a particular searching term, like, `isSearching` -> no NFTs were found based on your search. ???